### DISCUSSION TITLE
How to use functions from myst-cli with node.js?

### DISCUSSION THREAD

**Discussion by GYHHAHA** [executablebooks/meta#986] [Q&A] [answered]
I basically follow the steps by @rowanc1 from [myst-playground](https://github.com/rowanc1/myst-playground). In my code, I have `import { processNotebook } from "myst-cli";` and use it for process, however when I run the `npm run build`, it raise:

```
[build:myst] 
[build:myst] > myst-playground@0.0.0 build:myst
[build:myst] > esbuild myst.ts --bundle --platform=node --outfile=dist/myst.js
[build:myst]
[build:myst] X [ERROR] Could not resolve "p-limit"
[build:myst]
[build:myst]     node_modules/myst-cli/dist/cjs/transforms/links.js:54:37:
[build:myst]       54 │ var import_p_limit = __toESM(require("p-limit"));
[build:myst]          ╵                                      ~~~~~~~~~
[build:myst]
[build:myst]   You can mark the path "p-limit" as external to exclude it from the bundle, which will remove this error. You can also surround this "require" call with a try/catch block to handle this failure at run-time instead of bundle-time.
[build:myst]
[build:myst] 1 error
[build:myst] node:child_process:910
[build:myst]     throw err;
[build:myst]     ^
[build:myst]
[build:myst] Error: Command failed: C:\Users\gyhst\OneDrive\Desktop\Learn\React\my-node\node_modules\@esbuild\win32-x64\esbuild.exe myst.ts --bundle --platform=node --outfile=dist/myst.js
[build:myst]     at checkExecSyncError (node:child_process:871:11)
[build:myst]     at Object.execFileSync (node:child_process:907:15)
[build:myst]     at Object.<anonymous> (C:\Users\gyhst\OneDrive\Desktop\Learn\React\my-node\node_modules\esbuild\bin\esbuild:220:28)
[build:myst]     at Module._compile (node:internal/modules/cjs/loader:1159:14)
[build:myst]     at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
[build:myst]     at Module.load (node:internal/modules/cjs/loader:1037:32)
[build:myst]     at Module._load (node:internal/modules/cjs/loader:878:12)
[build:myst]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[build:myst]     at node:internal/main/run_main_module:23:47 {
[build:myst]   status: 1,
[build:myst]   signal: null,
[build:myst]   output: [ null, null, null ],
[build:myst]   pid: 8044,
[build:myst]   stdout: null,
[build:myst]   stderr: null
[build:myst] }
[build:myst]
[build:myst] Node.js v18.12.1
```
Thanks for any help!

**Comment by rowanc1** [comment, marked as answer]
It looks like we have missed `p-limit` as a dependency (it is brought in by some of the testing tools). I will get a PR up to fix this shortly and get a release out!

**Reply by rowanc1** [reply]
There should be a fix for this coming in here:
https://github.com/executablebooks/mystjs/pull/341

**Reply by rowanc1** [reply]
This is released in v0.1.22! Hopefully gets you up and running, if you do run into other things on packaging, don't hesitate to open issues! Glad that you found that playground repo. 🚀

**Reply by GYHHAHA** [reply]
Thanks! Now this works! 🚀

**Reply by rowanc1** [reply]
Wonderful, glad we got past that pretty easily, thanks for helping track this down!

**Reply by GYHHAHA** [reply]
I'm currently integrating many features from mystjs which is a wonderful tool for the people who would like to render myst content in client side. Since mystjs and myst-theme are still under beta version and formal doc is not fully-prepared, it's kind of tricky to handle everything well in both browser and node env. Thus I got many questions on the usage. Thanks again!

**Reply by rowanc1** [reply]
That is great! I am glad you are finding your way around a little bit at least, happy to continue pointing you to places and fixing things where we can!
